Remove libcontainer from containerd-shim

thaJeztah opened the pull request:
Libcontainer was only used for a single function, which in itself was just a tiny wrapper around `golang.org/x/sys/unix`; https://github.com/opencontainers/runc/blob/dbe44cbb1099eabfaa753322be2055a0a6e8fe2d/libcontainer/system/linux.go#L127-L130

```go
// SetSubreaper sets the value i as the subreaper setting for the calling process
func SetSubreaper(i int) error {
	return unix.Prctl(unix.PR_SET_CHILD_SUBREAPER, uintptr(i), 0, 0, 0)
}
```

This patch replaces it with the implementation in our sys package

thaJeztah commented:
@AkihiroSuda ptal

thaJeztah commented on cmd/containerd-shim/shim_linux.go:
```diff
@@ -45,3 +44,8 @@ func setupSignals() (chan os.Signal, error) {
 }
+
+// setSubreaper sets the value i as the subreaper setting for the calling process
+func setSubreaper(i int) error {
```
Guess I can even remove the argument, as it's a fixed value; let me update

fuweid replied:
please check here https://github.com/containerd/containerd/blob/master/sys/reaper_linux.go#L36

dims commented:
LGTM excellent find!

fuweid commented:
The `sys/reaper_linux.go` defines `SetSubreaper` function. What about using the `sys.SetSubreaper`?

thaJeztah commented:
@fuweid good catch;  Let me use that one

Looking at that package, it looks like it's still defining a custom const for `PR_SET_CHILD_SUBREAPER`; I'll open a PR for that as well

fuweid commented:
@thaJeztah LGTM.

thaJeztah commented:
Updated, and replaced it with the one in `sys`

Doing a follow-up to fix that implementation (and doing some slight refactoring to move it into the `sys/reaper` package perhaps)

dims commented:
LGTM
